core/src/main/java/cz/xtf/core/openshift/OpenShiftBinary.java

```diff
@@ -31,27 +34,32 @@ public void startBuild(String buildConfig, String sourcePath) {
 	}
 
 	// Common method for any oc command call
-	public void execute(String... args) {
-		executeCommand(ArrayUtils.addAll(new String[]{path}, args));
+	public String execute(String... args) {
+		return executeCommand(ArrayUtils.addAll(new String[]{path}, args));
 	}
 
 	// Internal
-	private void executeCommand(String... args) {
+	private String executeCommand(String... args) {
 		ProcessBuilder pb = new ProcessBuilder(args);
 
-		pb.redirectOutput(ProcessBuilder.Redirect.INHERIT);
-		pb.redirectError(ProcessBuilder.Redirect.INHERIT);
-
-		int result = -1;
+		pb.redirectOutput(ProcessBuilder.Redirect.PIPE);
+		pb.redirectErrorStream(true);
 
 		try {
-			result = pb.start().waitFor();
+			Process p = pb.start();
+			int result = p.waitFor();
```

Comment:
I suspect this can deadlock, if the process is writing more than the size of the buffer, as you are not consuming its stdout.

Reply:
you should probably read the InputStream in a separate thread instead after start.